Diablo Edit2/DlgCharItems.cpp

```diff
@@ -662,14 +663,40 @@ static void DrawGrid(CPaintDC & dc, const CRect & rect, int intervalX = 0, int i
 	}
 }
 
+void CDlgCharItems::CalculateScale()
+{
+	auto pWnd = GetDlgItem(IDC_CHECK1);	// This control should have an pixel left position of 338 and bottom position of 58
```

Comment:
Where do 338 and 58 come from?

Reply:
There is no RECT defining the drawing area for the different grids, so you can get the scale of the grid by the controls on the page.  The IDC_CHECK! control is in a good spot to determine what the scale is for both width and height.  The gird is supposed to be 30 x 30 pixels, and  all grids or equipped slots are a multiple of that.  So, you can calculate what the position of that control should have been to fit those controls without scaling (i.e., scale of 1.0).  This is where this number comes from.